lib/virt_autotest/utils.pm

```diff
@@ -64,6 +65,31 @@ sub is_xen_host {
     return get_var("XEN") || check_var("SYSTEM_ROLE", "xen") || check_var("HOST_HYPERVISOR", "xen") || check_var("REGRESSION", "xen-hypervisor");
 }
 
+# Usage: check_modular_libvirt_daemons([daemon1_name daemon2_name ...]). For example:
+# to specify daemons which will be checked: check_modular_libvirt_daemons(qemu storage ...)
+# to check all required modular daemons without any daemons passed
+sub check_modular_libvirt_daemons {
+    my @daemons;
+
+    if (@_ == 0) {
+        if (is_xen_host) {
+            @daemons = qw(xen network nodedev nwfilter secret storage);
+        } else {
+            @daemons = qw(qemu network nodedev nwfilter secret storage);
```

Comment:
Better to also add check for virtproxyd, virtlogd, virtlockd, for both kvm and xen.

Reply:
Super! try to add virtproxyd, virtlogd, virtlockd in the loop both on kvm and xen.

Reply:
Quick update,  after more investigation, figure out the **virtproxyd** and **virtlockd** disabled as the default:
```
     # systemctl status virtproxyd.service 
○ virtproxyd.service - Virtualization daemon
     Loaded: loaded (/usr/lib/systemd/system/virtproxyd.service; disabled; vendor preset: disabled)
     Active: inactive (dead)
TriggeredBy: ○ virtproxyd-admin.socket
             ○ virtproxyd-ro.socket
             ○ virtproxyd.socket
       Docs: man:virtproxyd(8)
             https://libvirt.org

# systemctl status virtlockd.service 
○ virtlockd.service - Virtual machine lock manager
     Loaded: loaded (/usr/lib/systemd/system/virtlockd.service; indirect; vendor preset: disabled)
     Active: inactive (dead)
TriggeredBy: ○ virtlockd-admin.socket
             ● virtlockd.socket
       Docs: man:virtlockd(8)
             https://libvirt.org
```

We'd better start or restart **virtproxyd** and **virtlockd** firstly and then check with them from func `check_modular_libvirt_daemons()`

Reply:
This function will be called by restart_modular_libvirt_daemons, right? I think the logic should be what are started/restarted in restart_modular_libvirt_daemons, are good to check status via your function here. 

With above guiding, virtproxyd is restarted in restart_modular_libvirt_daemons, while virtlockd is required by virtqemud/virtxend in their service file, so it should be active too. Then checking directly here will be ok, right?

At last,  just a reminder, above is a tiny thing I think. More focus can be given to the implementation of equivalent modular libvirt operations, for wherever have libvirtd operations, which at the moment only have implementations for monolithic libvirt. This involves both openqa and virtauto code, for all projects' code. Having a grep  for 'libvirt|qemu|xen' key words can quickly show an overview where have such TODO  items. Although it sounds like a lot, but actually in my impression, there won't be much todo actually.

Thanks for the effort. Look forward to the virtauto PR for modular libvirt!

Reply:
yes, this func will be called by restart_modular_libvirt_daemons().

Reply:
virtlockd and virtlogd are expected to be started by other libvirt daemons automatically in modular libvirt. we can check their status but I don't think they are required to be restarted in restart_modular_libvirtd_daemons(). Our automation tests should not interfere the natural behaviour of the testing target if not necessary. ie. if they would not be restarted automacally, it is a bug of libvirt which is expected to be detected by automation tests.

Reply:
Short update, after enabled virtqemud.service on the latest 15SP6 alpha build13.5 KVM host with Modular libvirt daemons, 
confirm that virtlockd.socket and virtlogd.socket start to be active as expected
FYI
```
# systemctl status virtqemud.service
● virtqemud.service - Virtualization qemu daemon
     Loaded: loaded (/usr/lib/systemd/system/virtqemud.service; enabled; vendor preset: disabled)
     Active: active (running) since Fri 2023-08-18 09:54:12 EDT; 9ms ago
TriggeredBy: ● virtqemud-admin.socket
             ● virtqemud.socket
             ● virtqemud-ro.socket
       Docs: man:virtqemud(8)
             https://libvirt.org
   Main PID: 21851 (virtqemud)
      Tasks: 19 (limit: 32768)
     CGroup: /system.slice/virtqemud.service
             └─ 21851 /usr/sbin/virtqemud --timeout 120
             
# systemctl status virtlockd.socket 
● virtlockd.socket - Virtual machine lock manager socket
     Loaded: loaded (/usr/lib/systemd/system/virtlockd.socket; enabled; vendor preset: disabled)
     Active: active (listening) since Fri 2023-08-18 07:53:51 EDT; 3h 19min ago
   Triggers: ● virtlockd.service
     Listen: /run/libvirt/virtlockd-sock (Stream)
     CGroup: /system.slice/virtlockd.socket

Aug 18 07:53:51 localhost systemd[1]: Listening on Virtual machine lock manager socket.

# systemctl status virtlogd.socket 
● virtlogd.socket - Virtual machine log manager socket
     Loaded: loaded (/usr/lib/systemd/system/virtlogd.socket; enabled; vendor preset: enabled)
     Active: active (listening) since Fri 2023-08-18 07:53:51 EDT; 3h 19min ago
   Triggers: ● virtlogd.service
     Listen: /run/libvirt/virtlogd-sock (Stream)
     CGroup: /system.slice/virtlogd.socket

Aug 18 07:53:51 localhost systemd[1]: Listening on Virtual machine log manager socket.
```
Meanwhile, virtlockd.service and virtlogd.service still keep inactive as the default, which should activate the service when needed.  double confirm this with Jim together.

Reply:
Had added to check for virtproxyd, virtlogd, virtlockd, for both kvm and xen with modular libvirt daemons.